upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template

```
@@ -109,14 +109,7 @@ spec:
           name: kube-router-cfg
       hostNetwork: true
       tolerations:
-      - key: CriticalAddonsOnly
-        operator: Exists
-      - effect: NoSchedule
-        key: node-role.kubernetes.io/master
-        operator: Exists
-      - effect: NoSchedule
-        key: node.kubernetes.io/not-ready
-        operator: Exists
+      - operator: Exists
```

I think we should go with what Weave has, which seems good enough. As both have similar main contributors could be easily accepted as a patch in upstream.
https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template#L251

```suggestion
      - effect: NoSchedule
        operator: Exists
      - effect: NoExecute
        operator: Exists
      - key: CriticalAddonsOnly
        operator: Exists
```

---

I don't see the practical difference between this and what Weave has. And the `CriticalAddonsOnly` one appears to be obsolete, replaced by the `system-node-critical` priorityClassName.

---

There is no practical difference from my point of view either, just easier to explain a PR for upstream.

---

The more succinct version seems clear enough. I'd suggest leading with that for an upstream PR.

---

OK